### PR TITLE
sdk: Extract slot-history sysvar

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

Just about all of the sysvars exist in separate crates, except for slot history.

#### Summary of changes

Extract it into a new crate. Looks pretty similar to slot hashes. All of the `info!` calls and logging dependencies were removed.

### CODE REVIEW

File: sdk/slot-history/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,206 @@
+//! A type to hold data for the [`SlotHistory` sysvar][sv].
+//!
+//! [sv]: https://docs.solanalabs.com/runtime/sysvars#slothistory
+//!
+//! The sysvar ID is declared in [`sysvar`].
+//!
+//! [`sysvar`]: crate::sysvar
```
Comment (joncinque) [inline sdk/slot-history/src/lib.rs]: This will get done in future work, once all the sysvar types are extracted

Comment (kevinheavey) [reply sdk/slot-history/src/lib.rs]: Let's update the doc link?

Comment (joncinque) [reply sdk/slot-history/src/lib.rs]: Sure, done

File: sdk/slot-history/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
+//! A type to hold data for the [`SlotHistory` sysvar][sv].
+//!
+//! [sv]: https://docs.solanalabs.com/runtime/sysvars#slothistory
+//!
+//! The sysvar ID is declared in [`sysvar::slot_history`].
+//!
+//! [`sysvar`]: https://docs.rs/solana-program/latest/solana_program/sysvar/slot_history
```
Comment (kevinheavey) [inline sdk/slot-history/src/lib.rs]: this should be ``[`sysvar::slot_history`]``

Comment (joncinque) [reply sdk/slot-history/src/lib.rs]: oops, that's right, thanks

**Review by kevinheavey — APPROVED** [review]

File: sdk/program/src/slot_history.rs [hunk 3]
```diff
@@ -1,211 +1 @@
...
-        assert_eq!(slot_history.oldest(), 0);
-        slot_history.add(10);
-        assert_eq!(slot_history.oldest(), 0);
-        slot_history.add(MAX_ENTRIES - 1);
-        assert_eq!(slot_history.oldest(), 0);
-        slot_history.add(MAX_ENTRIES);
-        assert_eq!(slot_history.oldest(), 1);
-    }
-}
+pub use {solana_clock::Slot, solana_slot_history::*};
```
Comment (febo) [inline sdk/program/src/slot_history.rs:2]: Should we add the deprecated message on this re-export?

Comment (joncinque) [reply sdk/program/src/slot_history.rs:2]: Done!

**Comment by febo** [comment]
Looks good - I think we are only missing the crate owner for CI.

**Comment by joncinque** [comment]
@yihau can you accept the ownership of `solana-slot-history`?

**Comment by yihau** [comment]
✅

**Comment by joncinque** [comment]
@febo can I get an approval to merge this?

**Review by febo — APPROVED** [review]

**Pull request merged**